keras_nlp/models/xlm_roberta/xlm_roberta_preprocessing.py

```diff
@@ -169,3 +159,87 @@ def from_preset(
         **kwargs,
     ):
         raise NotImplementedError
+
+
+class XLMRobertaTokenizer(SentencePieceTokenizer):
+    def __init__(
+        self,
+        proto,
+        sequence_length=None,
+    ):
+        super().__init__(proto=proto, sequence_length=sequence_length)
+
+    def vocabulary_size(self):
+        """Get the size of the tokenizer vocabulary."""
+        return super().vocabulary_size() + 1
+
+    def get_vocabulary(self):
+        """Get the size of the tokenizer vocabulary."""
+        vocabulary = tensor_to_string_list(
+            self._sentence_piece.id_to_string(
+                tf.range(self.vocabulary_size() - 1)
```

Comment:
Maybe just user `super().vocabulary_size()` here? Otherwise the increment/decrement gets hard to track.